webhook/env.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"os"
+	"strconv"
+)
+
+const portEnvKey = "WEBHOOK_PORT"
+
+// Port returns the webhook port set by portEnvKey, or default port if env var is invalid.
+func Port(defaultPort int) int {
+	port, err := strconv.Atoi(os.Getenv(portEnvKey))
+	if err != nil || port == 0 {
+		return defaultPort
```

Comment:
if `WEBHOOK_PORT` is set but not a valid port # we shouldn't fallback to the default port and instead return an error

ie. "9090 " (extra whitespace) will cause this to fallback to the default which is non-obvious

Reply:
Done. Thanks.